### ISSUE TITLE
fix(model/v41): fail closed at every unimplemented V4.1 forward stage

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12902]
# fix(model/v41): fail closed at every unimplemented V4.1 forward stage

```routing
lane: model
paths: ["internal/model/v41_failclosed_test.go"]
expected_steps: 3
repo: fak
public_issue: anthony-chaudhary/fak#12640
```

## Parent context
Parent: #12640 (native DeepSeek V4.1 Flash execution and physical qualification).

## Current state
Implementing this leaf's through-line is not yet started; the parent track is open.

## Problem frame
- Centrality: Core ? keeps partial forward work from leaking into generic paths.
- P1: advanced - keeps partial forward work from leaking into generic paths.
- P2: advanced - preserves the fail-closed invariant as leaves land.
- P3: preserved - compatible families unchanged.
- P4: advanced - negative admission tests.

## Why now
This leaf is on the critical path for native V4.1 execution and its parent track is open.

## Core through-line
Audit every loader and forward entrypoint that can receive a V4.1 config and assert it returns the
typed unsupported-native error rather than reaching a generic path. This hardens the guard landed
in `internal/model/v41_config.go` (commit `2d2bca9c`) as the forward leaves land out of order.

## Working spine
Intake -> bounded change -> focused witness -> landed receipt.

## Gold-plating boundary
No new forward capability. Negative/admission testing only. Do not weaken existing rejections or
change public loader signatures.

## Witness
```bash
go -C ..\fak test ./internal/model -run 'TestV41FailClosed' -count=1
```

### File:Line Seam
`internal/model/safetensors_quant.go:93` and `internal/model/weights.go:357`

### Blast Radius & Affected Lanes
- **Primary Lane:** `model`
- **Unaffected Lanes:** compatible model families unchanged

### Quarantined Fallback Mechanism
The typed error is the quarantine; trunk stays green with V4.1 execution held.

## Scoped Acceptance Criteria
- [ ] [SW-VERIFIED] Every f32 and quant loader entrypoint rejects V4.1 before weight I/O.
- [ ] [SW-VERIFIED] A partially implemented forward stage returns a typed error, not partial logits.
- [ ] [SW-VERIFIED] No test is skipped or weakened.

## Done condition
The scoped acceptance criteria above pass with the cited witness; no sibling lane regresses.

## Acceptance gate
Focused witness passes, package tests are green, vet and 5-gate boundary checks are clean.

## Work estimate
Estimate: 2 points (bounded leaf).

## Overall completion contribution
Contribution: 2/40 points (provisional baseline).

## Completion standard
development

## Closure binding
The resolving commit cites this issue with Fixes/Ref and carries the repository-qualified lane trailer.

**Comment by anthony-chaudhary** [comment]
Coordination request for the current #12902 worker: the issue routing contract names only `internal/model/v41_failclosed_test.go`, but the live `v41-failclosed-12902` lease covers `internal/model/**`. That broad scope blocks the already-admitted #12869 MTP continuation, whose five paths are `metal_mtp.go`, `specbind.go`, `metal_mtp_round_result_test.go`, `metal_mtp_test.go`, and `verify.go` under `internal/model/`.

Please reconcile your actual write scope and narrow the lease through the normal ownership mechanism to the contracted file if that is still the complete scope, or release it when your work is finished. If additional paths are necessary, record them here so we can coordinate disjoint work.

Observed lease: holder `opencode:ses_f82a1ff87ffeI6rKM102RDnFZH`, generation 1, object `3379c4bd6934a3f93d6d07f2cdd7f06065daef0e`, expiry Unix `1789242960`. Local session lookup did not find the owner; this is not evidence of abandonment. We have not changed the peer lease or begun overlapping writes.

**Comment by anthony-chaudhary** [comment]
﻿Landed on public `fak` main: **056f7ded22f24fc0bbdd8a6daa2f40f58568bc2e**

```
fix(model/v41): fail closed at every unimplemented V4.1 forward stage (fak model)
```

### What landed
Extends the V4.1 fail-closed invariant from the f32/`newModel` load gate to every seam that can receive a V4.1 config and otherwise reach a generic path:

- **QuantBuilder** (`NewQuantBuilder`/`AddF32Tensor`/`SetQ2KEmbedding`/`Build` + resident-quant targets) records a sticky `ErrV41NativeUnsupported`, closing the GGUF quant-on-load path (`internal/ggufload` via `QuantModel`/`QuantModelQ4K`) that never passes through `newModel`/`NewFromF32Tensors`.
- **`AddCanonicalMTPQ4K`/`AddCanonicalMTPFCQ8`** refuse V4.1 even when a crafted config also satisfies `isQwen35TextFamily` via a `linear_attention` LayerTypes entry.
- **`ClassifyForwardPath`** returns the typed error instead of classifying V4.1 as `ForwardAttnSeqGQA` and running a generic forward over V4.1 geometry.

Non-V4.1 families are unchanged. New `v41_failclosed_test.go` pins the refusal at each seam with negative controls that would fold weights if a guard were removed.

Note: the AWQ fence originally in this leaf was superseded by the independently-landed #12891 (`6b5442d52`, `LoadAWQ`/`LoadEXL2` fences); the rebase dropped the now-redundant hunk. This commit carries the remaining unique seams (QuantBuilder, canonical-MTP, forward classification).

### Witness
```
go test ./internal/model -run 'TestV41FailClosed|TestDeepSeekV41' -count=1
ok  	github.com/anthony-chaudhary/fak/internal/model
```
Also green: full `go test ./internal/model -count=1` (359s), `go vet ./internal/model`, 5-gate boundary check (`--staged`, 0 violations), leak audit clean.

### Files
- `internal/model/arch_support.go`
- `internal/model/quant_q4k.go`
- `internal/model/safetensors_quant.go`
- `internal/model/v41_failclosed_test.go`

Ref #12902
